cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -460,9 +476,53 @@ func (m *McmManager) GetMachineDeploymentNodeTemplate(machinedeployment *Machine
 		}
 		region = mc.Spec.Location
 		if mc.Spec.Properties.Zone != nil {
-			// Do not re-use the zone before re-vendoring mcm.
 			zone = mc.Spec.Location + "-" + strconv.Itoa(*mc.Spec.Properties.Zone)
 		}
+	case kindMachineClass:
+		mc, err := m.machineclient.MachineClasses(m.namespace).Get(context.TODO(), machineClass.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("Unable to fetch %s for %s, Error: %v", kindMachineClass, machinedeployment.Name, err)
+		}
+		switch mc.Provider {
+		case providerAWS:
+			var providerSpec *awsapis.AWSProviderSpec
+			err = json.Unmarshal(mc.ProviderSpec.Raw, &providerSpec)
+			if err != nil {
+				return nil, fmt.Errorf("Unable to convert from %s to %s for %s, Error: %v", kindMachineClass, providerAWS, machinedeployment.Name, err)
+			}
+
+			awsInstance := aws.InstanceTypes[providerSpec.MachineType]
+			instance = instanceType{
+				InstanceType: awsInstance.InstanceType,
+				VCPU:         awsInstance.VCPU,
+				MemoryMb:     awsInstance.MemoryMb,
+				GPU:          awsInstance.GPU,
+			}
+			region = providerSpec.Region
+			if mc.Labels != nil {
+				zone = mc.Labels[labelForFailureDomain]
+			}
+		case providerAzure:
+			var providerSpec *azureapis.AzureProviderSpec
+			err = json.Unmarshal(mc.ProviderSpec.Raw, &providerSpec)
+			if err != nil {
+				return nil, fmt.Errorf("Unable to convert from %s to %s for %s, Error: %v", kindMachineClass, providerAzure, machinedeployment.Name, err)
+			}
+
+			azureInstance := aws.InstanceTypes[providerSpec.Properties.HardwareProfile.VMSize]
+			instance = instanceType{
+				InstanceType: azureInstance.InstanceType,
+				VCPU:         azureInstance.VCPU,
+				MemoryMb:     azureInstance.MemoryMb,
+				GPU:          azureInstance.GPU,
+			}
+			region = providerSpec.Location
+			if mc.Labels != nil {
```

Comment:
Do we need this check? We are not reading the zone from the label like in the case of AWS.

Reply:
I guess we had the field for the zone on AzureMachineClass but not the AWSMachineClass and hence the label was a hack to accommodate this change. This is in line with the current in-tree code (not sure if it's the right way though) - https://github.com/gardener/autoscaler/pull/65/files#diff-fc87dde41e1a7abff0b0bc24dbabc258ac3d594a346b8f01e020b9e7f9775440R479.

Reply:
The field is here - https://github.com/gardener/machine-controller-manager/blob/master/pkg/apis/machine/v1alpha1/azure_machineclass_types.go#L101.

Reply:
I miss read your comment. I got the context. Let me fix this.

Reply:
Fixed it here - https://github.com/gardener/autoscaler/pull/65/commits/2df2c3fe482fd06a81ffda9f391d6552823c6c17. Sorry about that mistake.